active-record/src/main/java/com/iluwatar/active/record/MagicWand.java

```diff
@@ -0,0 +1,264 @@
+            ps.setString(4, this.getCore().name());
+
+            ps.execute();
+
+            ps.close();
+            connection.close();
+            return id;
+        }
+        catch (final SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void delete() {
+        validateToDelete();
+        try {
+            final Connection connection = getConnection();
+            final PreparedStatement ps = connection.prepareStatement(DELETE_SQL);
+
+
+            ps.setLong(1, this.getId());
+            ps.execute();
+            ps.close();
+            connection.close();
+        }
+        catch (final SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void update() {
+        validateToUpdate();
+        try {
+            final Connection connection = getConnection();
+            final PreparedStatement ps = connection.prepareStatement(UPDATE_SQL);
+
+
+            ps.setDouble(1, this.getLengthInches());
+            ps.setString(2, this.getWood().name());
+            ps.setString(3, this.getCore().name());
+            ps.setLong(4, this.getId());
+
+            ps.execute();
+            ps.close();
+            connection.close();
+        }
+        catch (final SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static MagicWand find(long id) {
+
+        try {
+            final Connection connection = getConnection();
+            final PreparedStatement ps = connection.prepareStatement(SELECT_SQL);
+
+            ResultSet rs;
+            ps.setLong(1, id);
+            rs = ps.executeQuery();
+            if (rs.next()) {
+                final MagicWand wand = new MagicWand();
+
+                wand.id = rs.getLong("id");
+                wand.setCore(WandCoreType.valueOf(rs.getString("core")));
+                wand.setWood(WandWoodType.valueOf(rs.getString("wood")));
+                wand.setLengthInches(rs.getDouble("length_inches"));
+
+                return wand;
+            }
+            else {
+                return null;
+            }
+        }
+        catch (final SQLException e) {
+            throw new RuntimeException(e);
+        }
+
+    }
+
+    private void validateToSave() {
+        validateFieldsToBeNotNull();
+        if (this.getId() != null)
+            throw new IllegalStateException("Can not save wand that was previously saved. Use 'update' metod instead.");
+    }
+
+    private void validateToUpdate() {
+        validateFieldsToBeNotNull();
+        if (this.getId() == null) throw new IllegalStateException("Can not update a record without ID specified");
+    }
+
+    private void validateToDelete() {
+        if (this.getId() == null) throw new IllegalStateException("Can not delete a record without ID specified");
+    }
+
+    private void validateFieldsToBeNotNull() {
+        if (this.getLengthInches() == null)
```

Comment:
@stephen-lazarionok A `MagicWand` will not be created only to be saved in database. What if a wand is created to cast a fireball? In that scenario we allowed an invariant that is core should not be `null ` and when `castFireball()` will be called it will say "Not enough magic power" but actually the wand is not yet built. I hope you understand my concern. But again these are all nitpicks. We can even utilize `Builder` based #184 for creating a wand if constructor is getting too long. But let's not over-engineer :smiley: